src/jvm/main/org/jetbrains/kotlinx/lincheck/tracerecorder/TraceCollectingEventTracker.kt

```diff
@@ -75,7 +74,7 @@ private class ThreadData(
 }
 
 enum class TraceCollectorOutputType {
-    BINARY, TEXT, VERBOSE
+    BINARY, `BINARY-MEM`, TEXT, VERBOSE
```

Comment:
What does `BINARY-MEM` stands for?

Why not simply `BINARY_MEM`?

Reply:
> BINARY-MEM is strategy to collect-and-dump, as opposite to BINARY which is «streaming» API

Left a comment below with a proposal.

>  I don't like _ in configs, so I choose to use -.

By configs, you mean the way it is passed from CLI, as `binary-mem` ?
If so, then I think the enum entry name is not necessarily should correspond to the way it is passed from CLI or other configs. For instance, you can just modify the `toTraceCollectorOutputType()` method implementation, or override `TraceCollectorMode.toString()`, etc. 

IMO, for enum class name it is better to use underscore `BINARY_MEM`, because otherwise we need to wrap the name into quotation everywhere, which is less convenient.